### ISSUE TITLE
Join cluster made  other nodes already in cluster crashed

### ISSUE THREAD

**Issue by ForLina** [emqx/emqx#1407]
#### Environment

- OS: CentOS 7
- Erlang/OTP: 18.3
- EMQ: 2.0.7

#### Description

We have 7 nodes with 8C16G running on aliCloud which already in a cluster.
Yestoday I tried to join a new empty node into cluster, It failed and soon other two nodes in cluster 
crashed. A Mnesia core generated，and the crash info in Mnesia core is：
`{crashinfo,{"~p~n",
            [{error,{"Cannot open dets table",schema,
                     [{file,"/home/admin/emqttd/data/mnesia/emqttd@10.45.150.72/schema.DAT"},
                      {keypos,2},
                      {repair,true},
                      {type,set}],
                     {file_error,"/home/admin/emqttd/data/mnesia/emqttd@10.45.150.72/schema.DAT",
                                 enoent}}}]}}`
 
This hanppend more than once. 
At first we don't have many clients, join into a cluster always succeed, but now we have about 100k connections on each node, and the join operation almost failed every time.
Hope you can fix this issue.

**Comment by HJianBo** [comment]
It's seems the file `./data/mnesia/emqttd@10.45.150.72/schema.DAT` had gone

Have you check that?

**Comment by ForLina** [comment]
I don't really member , but nobody operated this server at that time which means the file can't be deleted by us.

**Comment by ForLina** [comment]
A crash of emq cluster happened when backbone network was not stable(optical fibre delivery or something else).
I checked the system log and found that the beam process is not kill by oom_killer, in system message log, I found an error 
/var/log/messages
`Dec 31 18:04:58 mqtt-real3 run_erl[7792]: errno=2 'No such file or directory'#012Error in reading from terminal
Dec 31 18:04:59 mqtt-real3 kernel: TCP: too many orphaned sockets
Dec 31 18:04:59 mqtt-real3 kernel: TCP: too many orphaned sockets
Dec 31 18:04:59 mqtt-real3 kernel: TCP: too many orphaned sockets
Dec 31 18:04:59 mqtt-real3 kernel: TCP: too many orphaned sockets
Dec 31 18:04:59 mqtt-real3 kernel: TCP: too many orphaned sockets
Dec 31 18:04:59 mqtt-real3 kernel: TCP: too many orphaned sockets
Dec 31 18:04:59 mqtt-real3 kernel: TCP: too many orphaned sockets
Dec 31 18:04:59 mqtt-real3 kernel: TCP: too many orphaned sockets
Dec 31 18:04:59 mqtt-real3 kernel: TCP: too many orphaned sockets
Dec 31 18:04:59 mqtt-real3 kernel: TCP: too many orphaned sockets`

And no import log in crash.log or error.log.
I wonder if you have seen the same problem, or what may cause this happen since I can't reappear this problem in test environment.
